### DISCUSSION TITLE
[Feature Request][`/vault/{filename}` PUT/POST] Create folder when not exist

### DISCUSSION THREAD

**Discussion by SettingDust** [coddingtonbear/obsidian-local-rest-api#43] [Ideas]
Current: 500 Internal Server Error
Expect: Create the folder and file.

**Comment by SettingDust** [comment]
It's a `enhancement` just like #26. Why should be a discussion?

**Reply by coddingtonbear** [reply]
In general, enhancement requests should start as discussions so nobody wastes their time working on a thing that won't be merged.  Not all enhancement ideas are things that anybody is willing to work on themselves, and even the ones that _are_ might not be features that I'm willing to maintain long-term.  By starting as a discussion, there can be a back-and-forth dialog about whether the feature is desirable at all, and, if so, how the feature should work were it to be implemented.  Additionally, this discussion gives contributors an opportunity to decide who is going to take on the work of implementing the feature.

In the case of #26 -- that's a feature I've accepted as one I personally want to add to the project.  It has made it through the vetting process above, and I see it having enough value that I'm willing to maintain it in the future.  By making it an "Issue", I'm communicating that that feature is being worked on or will be worked on at some point in the future.
